src/perf/isolates.md

```diff
@@ -0,0 +1,286 @@
+
+The `Isolate.run` method requires a single argument, a callback function, that
+is run on the new isolate. This callback’s function signature must have exactly
+one required, unnamed argument. When the computation completes, it returns the
+callback’s value back to the main isolate, and exits the spawned isolate.
+
+For example, consider this code that loads a large JSON blob from a file, and
+converts that JSON into custom Dart objects. If the json decoding process wasn’t
+off loaded to a new isolate, this method would cause the UI to become
+unresponsive for several seconds.
+
+<?code-excerpt "lib/main.dart (isolate-run)"?>
+```dart
+// Produces a list of 211,640 photo objects.
+// (The JSON file is ~20MB.)
+Future<List<Photo>> getPhotos() async {
+  String jsonString = await rootBundle.loadString('assets/photos.json');
+
+  final dynamic photos = await Isolate.run(() {
+    final photoData = jsonDecode(jsonString);
+    return photoData.map((dynamic element) {
+      final data = element as Map<String, dynamic>;
+      return Photo.fromJson(data);
+    }).toList();
+  });
+
+  return photos as List<Photo>;
+}
+```
+
+For a complete walkthrough of using Isolates to parse JSON in the background, see [this cookbook recipe][].
+
+[this cookbook recipe]: {{site.url}}/cookbook/networking/background-parsing
+
+### Isolates on the web
+
+Dart doesn’t support isolates on web platforms. If you’re targeting the web with
+your Flutter app, you should use the compute method. The [`compute()`][] method
+runs the computation on the main thread on the web, but spawns a new thread on
+mobile devices. On mobile and desktop platforms `await compute(fun, message)` is
+equivalent to `await Isolate.run(() => fun(message))`.
+
+## Stateful, longer-lived isolates
+
+- [`compute()`][] abstracts a handful of lower-level, isolate-related APIs to  simplify isolate management:
+- [`Isolate.spawn()`][] and [`Isolate.exit()`][]
+- [`ReceivePort`][] and [`SendPort`][]
+- [`send()`][] method
+
+When you use the compute method, the new isolate immediately shuts down after it
+returns a single message to the main isolate. Sometimes, you’ll need isolates
+that are long lived, and can pass multiple messages to each other over time. In
+Dart, you can accomplish this with the Isolate API and Ports. These long-lived
+isolates are colloquially known as _background workers_.
+
+Long-lived isolates are useful when you have a specific process that either
+needs to be run repeatedly throughout the lifetime of your application, or if
+you have a process that runs over a period of time and needs to yield multiple
+return values to the main isolate.
+
+### ReceivePorts and SendPorts
+
+Set up long-lived communication between isolates with two classes (in addition
+to Isolate): [`ReceivePort`][] and [`SendPort`][]. These ports are the only way isolates can
+communicate with each other.
+
+Ports behave similarly to Streams, in which the StreamController or Sink is
```

Comment:
Generally, if we uppercase the class and are talking about a specific class, we code font it. If we don't uppercase it, and are using the word generically, we don't code font it.

```suggestion
`Ports` behave similarly to `Streams`, in which the `StreamController` or `Sink` is
```